### DISCUSSION TITLE
Meta ideas for org perception

### DISCUSSION THREAD

**Discussion by jcbhmr** [tinylibs/rfcs#3] [General]
Hello @Aslemammad! I see you're the Owner of this @tinylibs organization. This is a discussion on some things that I think could take this organization "to the next level". But there's also some questions of whether or not you _want_ that and what your goals and vision for this organization were.

I want to stress that some of these improvements are contingent upon _what your vision for this @tinylibs organization is/was_ and that these are my own ideas.

Questions (some I can assume the answer to, but it's good to have an authoritative answer):
- What's the purpose of this organization?
- Is it your personal "namespace organization" where you dump a bunch of projects that you don't want under https://github.com/Aslemammad/*?
- Is it a business? I note that you have "founder of @tinylibs" in your GitHub profile
- Is it a community-driven repository that exclusively curates tiny packages?
- Is it just a @unjs lookalike with a different name?
- Is it JavaScript/TypeScript -only organization?
- Is there a "your package must be under N kilobytes at all times" rule?
- Are there lint rules for the organization?
- How does it compare to a group like @unjs ?
- How does it compare to a group like @un-ts ?
- How does it compare to a group like @jazzband ?
- How do new members get selected?
- What happens if the Owner looses interest?
- What's the governance structure (ie. who makes big decisions of org-related or project-related stuff)?
- What's the criteria for a new package to be made under the github.com/tinylibs/* namespace?
- How are packages added?
- Do packages need the tiny* naming convention?

Bear in mind that **you don't need to provide answers to all these questions in an essay** just casual "yeah I like this make it like that, but this way" is a pretty good guide. So for instance, if I were to start a @nodefill organization, I'd put the "mission statement" something like "make it like @jazzband, but for Node.js-related feature polyfills instead of Python/django"

If you were to ask me "what is @tinylibs?" I'd say: "It's Aslemammad's collection of smallish TypeScript projects; a front for https://npm.im/tinybench and https://npm.im/tinypool." 🤪
👆 I don't know if that's the identity that you want to portray? Or maybe it is. Hence this discussion 😉
![image](https://github.com/tinylibs/rfcs/assets/61068799/f460e4ba-b7d9-4680-9f9c-aacd30803bc1)

On an related note, here's some ideas for things to improve the meta-magic of this repository to take advantage of GitHub features and such. Basically just "shiny paint" 🤓
- Move this https://github.com/tinylibs/rfcs repo to https://github.com/tinylibs/.github since that's the GitHub mmeta repo which can house:
  - PR templates that apply to all repos that don't override them
  - Issue templates that apply to all repos that don't override them
  - CODE_OF_CONDUCT.md doc that apply to all repos that don't override them
  - CONTRIBUTING.md doc that apply to all repos that don't override them
  - SECURITY.md
  - SUPPORT.md
  - FUNDING.yml which can be overridden per-repo if a specific user wants to advertise their $$ donations stuff on a repo of their own that's under the @tinylibs umbrella
  - profile readme that shows up when you visit https://github.com/tinylibs and usually says something like "hey there! here's what this org is and our top things"
- put some kind of "mission statement" in the profile/README.md of the .github meta repo
- Add at least one other Owner person so that if you go on vacation or something someone else can make Security-related stuff happen (**probably one of the existing members that have been here for a while; not me 😅**) -- GitHub themselves recommend at least 2x Owner peoples. (see below)
- delete the https://github.com/tinylibs/tinywebpool repo (I can't since I'm not an Owner/Admin of that repo)
- archive this repo https://github.com/tinylibs/tinybench.old or delete it or something since it's not maintained? It's named ".old" so that's why I assume that it should be archived, but if it's just a stable fork, then it might want to be renamed 🤣

![image](https://github.com/tinylibs/rfcs/assets/61068799/576a0972-5cfa-420f-b3d8-1fa71c596d3d)

If you want to take it "to the next level" like I alluded to at the top, I think following something like either @jazzband or @unjs would be a good idea, but with a **scope that doesn't compete with theirs directly.** For instance, something like: "A community of JavaScript-ers who love tiny libraries. [Join], [Add your project]" or something would be cool. Basically **make it easier for people to participate** and **emphasize your niche/features/what you're org focuses on**. ex: @jazzband is Python/django, @unjs is js _tooling_.

Another idea that would be cool is to see a website kinda like https://unjs.io/ but for @tinylibs!

I hope that this comes across not as **criticism** but as **ideas** 💡. I think that you really have potential in the @tinylibs name to make it something really cool! Especially with the nice `tiny${thing}` prefix on all the libraries (though idk how well that will last)

**Comment by Aslemammad** [comment]
That's an excellent start for making tinylibs more community-based.

- A place for tiny and minimal projects (also dependency free as possible
- Nope, it's only space for minimal projects, it doesn't need to be mine or not (It's not like a place where I throw packages that I don't love, it's like a place for throwing more organized minimal projects, that need proper maintaining)
- No, it's not a business (but I love monetization), and the founder means started in that terms, I'm not planning to make it a business. But I'd like to keep it one founder/owner and many admins/contributors, each one controlling their own project with full control. As an owner, I manage to see which projects are proper for this place, give their maintainers full control and manage tinylibs as a whole.
- Yes, it's community-driven. 
- Nope, it's not like @unjs or any other place, though they can be a good inspiration since I use their packages so much. The plan is for tiny packages since the javascript ecosystem is so much bloated, and we want to bring the do-one-thing linux philosophy. 
- Any member is welcome with this condition
  - They have a tiny library in tinylibs
  - They maintain a library and have active maintenance, this would be discussed with the main maintainer though.
- I don't know how it'd be compared, our niche is "Tiny, minimal and dependency-free packages."
- > What happens if the Owner looses interest?

![image](https://github.com/tinylibs/rfcs/assets/37929992/4e63746a-75ce-4cf0-a53c-ca00eba7aa7d)

I guess we'll never know
- Org related: Owner, project related: Maintainers. At some point, we should have a discord channel for tinylibs so we bring maintainers and folks there. A special channel for maintainers would be good so we all discuss things there.
- archived the old tinybench just now and deleted tinywebpool
- > A community of JavaScript-ers who love tiny libraries. [Join], [Add your project]

I love this
- > make it easier for people to participate and emphasize your niche/features/what you're org focuses on

That'd be amazing.

- I'm the CEO of @pheno-agency, I'll let them know so we design a website for tinylibs. Let me know what you'd like it to have. Message me on discord: Aslemammad#6670 

Your ideas are amazing, If they can be done, tinylibs would take another step toward its mission. I believe in speed, if we achieve this stuff sooner, the better the outcome would be. We needed someone like you.

- > Move this https://github.com/tinylibs/rfcs repo to https://github.com/tinylibs/.github since that's the GitHub meta repo which can house:

done, it'd be good to have those files also.

**Reply by jcbhmr** [reply]
> What happens if the Owner looses interest?
> _I guess we'll never know_

While I get that you're trying to affirm your commitment... You might happen to: go on vacation, get sick, change interests, get a job somewhere demanding "no OSS", face legal challenges, move to another country, lose internet access due to a natural disaster, or experience any other unexpected event. Even [core-js](https://github.com/zloirock/core-js), the most popular JS polyfill monorepo [had a hiatus while the owner was incapacitated](https://github.com/zloirock/core-js/blob/master/docs/2023-02-14-so-whats-next.md#accident).

My recommendation is to follow the GitHub org banner thing and add one of the other contributors of tinybench or other long-time projects as top-level owners so that they can do things related to security and such if you're ever incapacitated.

![image](https://github.com/tinylibs/.github/assets/61068799/0e4298af-4573-46b9-8821-31808db2f30d)

<sup>Sidenote: I think this is a bit ironic: "Yes, it's community-driven." &nbsp; "one founder/owner...As an owner, I manage to see which projects are proper for this place" 🤣</sup>

**Reply by Aslemammad** [reply]
I agree with you! I wanted to make the community drive the packages instead of the whole organization (I'd like to be the one taking decisions on the org, and the maintainers on the package). Anyways, let's take your and github's suggestion. I made @Sajad-Sharhani a new owner. Let's see what happens, what other things you suggest, let me know, I'd like to hear.

**Reply by Aslemammad** [reply]
I'd like to add you also to one of the owners later, but let's see how it goes with your packages and management. 😊

**Comment by jcbhmr** [comment]
I made a very dead-simple Vuejs+Vite website that I think is sufficient for now. In the future it can be expanded on. 
![image](https://github.com/tinylibs/.github/assets/61068799/8e90cb65-a5e6-4b6c-b836-8af4c9edbff7)
https://github.com/jcbhmr/tinylibs.github.io
https://jcbhmr.github.io/tinylibs.github.io/

<sup>yes it's based heavily on the https://unjs.io/ website</sup>

**Reply by Aslemammad** [reply]
It's really nice, I wanted to make it more complex to have docs and more complex UI/UX, but let's go with this now.
